packages/react/src/components/fileInput/FileInput.tsx

```diff
@@ -0,0 +1,365 @@
+import React, { ChangeEvent, useEffect, useRef, useState } from 'react';
+
+// import core base styles
+import 'hds-core';
+import comboseAriaDescribedBy from '../../utils/comboseAriaDescribedBy';
+import classNames from '../../utils/classNames';
+import { Button } from '../button';
+import { IconPlus, IconPhoto, IconCross, IconDocument, IconUpload } from '../../icons';
+import { InputWrapper } from '../../internal/input-wrapper/InputWrapper';
+import styles from './FileInput.module.scss';
+
+type FileProperty = keyof File;
+
+const isEqualFileBy = (givenProperties: FileProperty[], a: File, b: File): boolean => {
+  const sameProps: FileProperty[] = givenProperties.filter((property: FileProperty) => a[property] === b[property]);
+  return sameProps.length === givenProperties.length;
+};
+
+const findDuplicateByNameAndType = (files: File[], fileToCompare: File): File | undefined =>
+  files.find((file: File) => isEqualFileBy(['name', 'type'], file, fileToCompare));
+
+type DragAndDropProps = {
+  label: string;
+  helperText: string;
+};
+
+type Language = 'en' | 'fi' | 'sv';
+
+type FileInputProps = {
+  /**
+   * The id of the input element
+   */
+  id: string;
+  /**
+   * The label for the input
+   */
+  label: string;
+  /**
+   * The label for the file button. The button is not visible for assistive technology
+   */
+  buttonLabel: string;
+  /**
+   * The language of the component. It affects which language is used to present component-specific messages, labels, and aria-labels
+   *
+   * @default "en"
```

Comment:
Should the default language be fi instead of en? 🤔

Reply:
- [ ] set "fi" as default lang

Reply:
We should probably change it to the comment too. I saw you changed it already to code, nice 👍

Reply:
- [ ] in documentation too